src/components/JournalForm/JournalFormEntryLines.tsx

```diff
@@ -32,15 +42,59 @@ export const JournalFormEntryLines = ({
     name: string,
     value: string | BaseSelectOption | number | undefined,
     lineItemIndex: number,
-    accounts?: LedgerAccountBalance[] | undefined,
+    accounts?: LedgerAccountBalance[],
   ) => void
   sendingForm: boolean
   config: JournalConfig
 }) => {
-  const { data: accountsData } = useContext(ChartOfAccountsContext)
+  const { data } = useAllCategories()
   const { form } = useContext(JournalContext)
 
-  const parentOptions = useParentOptions(accountsData)
+  const { flattenedCategories, parentOptions } = useMemo(() => {
+    const flattenedCategories = recursiveFlattenCategories(data ?? [])
+
+    const parentOptions = flattenedCategories
+      .sort((a, b) => (a.display_name.localeCompare(b.display_name)))
+      .map(({ display_name, id }) => {
+        return {
+          label: display_name,
+          value: id,
+        }
+      })
+
+    return { flattenedCategories, parentOptions }
```

Comment:
NIT: I can imagine that flattening categories could be a common thing to do across the application, so it could be moved to `src/utils/categories.ts`